### PR TITLE
Tests: 100% test coverage in utils.download

### PR DESCRIPTION
**Pull request by SiQube**
we might want to refactor this later to use mocks.

### CODE REVIEW

**Comment by SiQube** [comment]
this PR fixes #185

**Comment by SiQube** [comment]
rebase after #244

**Review by dkrako — CHANGES_REQUESTED** [review]
Thanks for this work!

I just have some minor change requests then we're ready to merge.

File: src/pymovements/utils/downloads.py [hunk 1]
```diff
@@ -77,6 +77,8 @@ def download_and_extract_archive(
     if extract_dirpath is None:
         extract_dirpath = download_dirpath
 
+    # mypy is unaware that 'extract_dirpath' can never be None (l.77)
```
[the remark this answers is not in the document]
Comment (SiQube) [reply src/pymovements/utils/downloads.py:80]: done

File: tests/utils/test_downloads.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import pytest
+
+from pymovements.utils.archives import extract_archive
+from pymovements.utils.downloads import _DownloadProgressBar
+from pymovements.utils.downloads import _get_redirected_url
+from pymovements.utils.downloads import download_and_extract_archive
+from pymovements.utils.downloads import download_file
+
+
+@pytest.fixture(name='test_dir')
```
[the remark this answers is not in the document]
Comment (SiQube) [reply tests/utils/test_downloads.py:38]: done

File: tests/utils/test_downloads.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+    with pytest.raises(RuntimeError) as excinfo:
+        download_and_extract_archive(url, tmp_path, download_filename, extract_dirpath, md5)
+
+    msg, = excinfo.value.args
+    assert msg == f"File {os.path.join(tmp_path, 'pymovements-0.4.0.tar.gz')} "\
+        'not found or download corrupted.'
+
+
+def test_extract_archive(tmp_path):
```
[the remark this answers is not in the document]
Comment (SiQube) [reply tests/utils/test_downloads.py]: done

File: src/pymovements/utils/downloads.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkrako) [reply src/pymovements/utils/downloads.py:80]: alright, so now only the comment is left over

**Review by dkrako — APPROVED** [review]
thx a lot!

**Pull request merged**